sdk/storage/Azure.Storage.DataMovement/src/StorageResourceItemProperties.cs

```diff
@@ -14,22 +14,22 @@ public class StorageResourceItemProperties
         /// <summary>
         /// The length of the resource.
         /// </summary>
-        public long? ResourceLength { get; }
+        public long? ResourceLength { get; set; }
```

Comment:
Why do these need to be settable now?

Reply:
see this I just made
https://github.com/Azure/azure-sdk-for-net/pull/45398/files#r1712077627

So I had to make this change cause the length wasn't being stored correctly, therefore not correctly seeing that there is a length in fact for downloading the file.